tests/src/Interop/StringMarshalling/Native/StringMarshalingNative.h

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <platformdefines.h>
+#include <type_traits>
+#include <algorithm>
+
+template<typename StringT>
+struct StringTraits
+{
+    using type = StringT;
+    using CharT = typename std::remove_pointer<StringT>::type;
+    using ConstStringT = CharT const *;
+    static const bool VerifyNullTerminator = true;
```

Comment:
I don't fully understand why `VerifyNullTerminator` exists?

Reply:
`BSTR`s are null terminated when used as strings, they just don't have to be. All `BSTR` allocating string functions append a null I believe.

Reply:
I think I added that as part of something that I've since changed. I'll go remove that.